### ISSUE TITLE
Black screen while playing videos

### ISSUE THREAD

**Issue by Noman-Aziz** [ente-io/ente#3744]
### Description

Hi,

I just started using your services and it is great. I'm having an issue both on the website and the linux desktop app. Basically, when i open any video (mp4), it first gets downloaded and then when playing, i can only hear the audio and see black screen on the video. I don't know what is causing it, I've attached the debug logs for your convenience.

Additionally, one other thing i wanted to ask was that each time i close and open the app, I have to redownload the same video in order to watch it. Is this expected behavior? Can there certain catching mechanism be implemented?

Looking forward to hearing from you guys, Thanks!

---

## Debug Logs

```
[2024-10-17T16:12:39+05:00] [main] [info] Starting ente-photos-desktop 1.7.5
[2024-10-17T16:12:39+05:00] [main] [info] Running on {
  platform: 'linux',
  osRelease: '6.11.3-arch1-1',
  systemVersion: '6.11.3-arch1-1'
}
[2024-10-17T16:12:39+05:00] APPIMAGE env is not defined, current application is not an AppImage
[2024-10-17T16:12:39+05:00] [main] [error] Failed to check for updates
[2024-10-17T16:12:40+05:00] [rndr] [info] Starting ente-photos-web git 9515cf7 uid 1580559962489395
[2024-10-17T16:12:40+05:00] [rndr] [info] DownloadManager already initialized
[2024-10-17T16:12:40+05:00] [rndr] [info] Folder watch deduced 0 events
[2024-10-17T16:12:49+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: false fetching: false
[2024-10-17T16:12:49+05:00] [rndr] [info] [93514329] new file src request
[2024-10-17T16:12:49+05:00] [rndr] [info] download attempted for file id 93514329
[2024-10-17T16:12:49+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: false fetching: true
[2024-10-17T16:12:49+05:00] [rndr] [info] [93514329] file download already in progress
[2024-10-17T16:13:02+05:00] [rndr] [info] [93514329] calling invalidateCurrItems for src, source loaded: true
[2024-10-17T16:13:02+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: true fetching: true
[2024-10-17T16:13:02+05:00] [rndr] [info] [93514329] source already loaded
[2024-10-17T16:13:02+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: true fetching: true
[2024-10-17T16:13:02+05:00] [rndr] [info] [93514329] source already loaded
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: false sourceLoaded: false fetching: false
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] doesn't have thumbnail
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: false sourceLoaded: false fetching: false
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] thumb download already in progress
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] calling invalidateCurrItems for thumbnail msrc: true
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: false fetching: false
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] new file src request
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: false fetching: true
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] file download already in progress
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] file download already in progress
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] calling invalidateCurrItems for src, source loaded: true
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: true fetching: true
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] source already loaded
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] getSlideData called for thumbnail: true sourceLoaded: true fetching: true
[2024-10-17T16:13:10+05:00] [rndr] [info] [93514329] source already loaded
```

### Version

v1.7.5

### What product are you using?

Ente Photos

### What platform are you using?

Desktop - Linux

**Comment by mnvr** [comment]
> when i open any video (mp4), it first gets downloaded and then when playing, i can only hear the audio and see black screen on the video

This sounds like a codec issue. On the desktop app, we ask the embedded chrome browser in the app if it can play the video, and if it says it can, then we don't try to unnecessarily convert it using ffmpeg. However, in some cases depending on the OS/codecs, part of the video is playable (e.g. the audio stream) so the pre-flight check causes the conversion to be skipped but only one stream is playable. This causes either the video to start playing without audio, or the audio to start playing without video.

As a fix for this (and for other issues), we are exploring the option to create an alternative version of videos with a standard codec so that it can be played back always irrespective of the source codecs (The details are not finalized yet)

> Can there certain catching mechanism be implemented?

Yes, this is something we hope to do in the longer term. The app earlier had a form of persistent on-disk caching, but that caused too much disk usage for some people, so we now use the embedded browser cache, but that doesn't always cache videos. One option we're thinking of is reusing the local export folder as a cache (this gives the benefit of the cache without any extra disk space usage).

---

ps. Thank you for the kind words!

**Comment by teilzeitutopist** [comment]
Same problem here with Fedora Linux and mp4 Videos. Does your explanation mean, I could do anything like install Codecs or Google Chrome on my system?

**Comment by mnvr** [comment]
We are currently working on the ability to create a streamable variant of videos that can be played without downloading the original. During this generation, we will also convert to a more compatible codec. The desktop app already has the ability to playback these streams, and support for generating them is being worked on. Tentatively, it is expected to be complete by the next release, 1.7.13 (the release notes will mention if that is the case).

tl;dr; next version might solve your issue

**Comment by Varen04** [comment]
so ... since we're on 1.7.14 and still having the issue, is this not implemented yet or is the new implementation still not working properly?

**Comment by mnvr** [comment]
The new implementation will only help if the Preferences > Streamable videos has been enabled and the stream for that particular video has been generated. More details here - https://help.ente.io/photos/faq/video-streaming

**Comment by codethief** [comment]
I'm having the same issue and I suspect it might be because the original video was taken by my phone in 10-bit HDR mode. When downloading the video file to my desktop machine (NixOS 25.11 with Xserver) the colors look washed out, suggesting that the available bit depth is not 10 bits. So maybe this is why Chrome just shows a black screen(?)
